### DISCUSSION TITLE
Query parameters for Oauth

### DISCUSSION THREAD

**Discussion by kevinlinxc** [reflex-dev/reflex#1037] [General]
Sometimes, your site has something like mysite.com/?param1=hi&param2=bye

In Streamlit, I can get and set these parameters with these functions:
https://docs.streamlit.io/library/api-reference/utilities/st.experimental_get_query_params
https://docs.streamlit.io/library/api-reference/utilities/st.experimental_set_query_params

Is there a way to do this in Pynecone? This is a really important feature to have because oauth authentication flows usually redirect you back to your site with a code in the url.

**Comment by jq6l43d1** [comment]
I use Azure AD for authentication on my website. I use get_query_params() https://pynecone.io/docs/components/pages It returns a dict containing each query. There are some other functions available too: https://github.com/pynecone-io/pynecone/blob/592be487c9605e899718c8642affc7cfc845e3a7/pynecone/state.py

**Reply by kevinlinxc** [reply]
@jq6l43d1 I noticed that in a PR when I searched for this, but all the examples seem to have /a/b format. Are you saying it also works for ?a=b use case?

**Reply by jq6l43d1** [reply]
![image](https://github.com/pynecone-io/pynecone/assets/110377310/bab4e2bd-e660-4ad7-8a61-64f03e705583)
get_current_page() will give you the path of the url.
get_query_params() will give you the query string of the url.

**Reply by kevinlinxc** [reply]
Is there a way to set the query parameters, e.g. to ingest and then get rid of a query parameter?
